.devcontainer/advanced-integration/devcontainer.json

```diff
@@ -0,0 +1,39 @@
+// For more details, see https://aka.ms/devcontainer.json.
+{
+	"name": "Advanced Integration",
+	"image": "mcr.microsoft.com/devcontainers/javascript-node:0-20",
+	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}/advanced-integration",
+
+	// Use 'onCreateCommand' to run commands when creating the container
+	"onCreateCommand": "if [ -n \"$CODESPACES\" ]; then sudo cp ../.devcontainer/advanced-integration/welcome-message.txt /usr/local/etc/vscode-dev-containers/first-run-notice.txt; fi;",
```

Comment:
Do we really need to condition this on `$CODESPACES`? It would be nice to keep the devcontainer simpler and seems like we should be able to remove this.

Reply:
If we remove this, then the welcome message would also show up when we create the container (locally) with Dev Containers extension.

As the instructions for both differ (mainly due to declarative secrets), added the condition.

Also to avoid this - I tried to copy the msg to `/workspaces/.codespaces/shared/first-run-notice.txt` instead of `/usr/local/etc/vscode-dev-containers/first-run-notice.txt`. But, that didn't help, I still saw the default prompt.

Reply:
I'd still rather have a configuration that works in both Codespaces and Dev Containers. Can we just change the welcome message to not say "In GitHub Codespaces"?

We also can't assume secrets are already setup in Codespaces, even with declarative secrets. Declarative secrets are optional and users can proceed without them. Therefore, in both Codespaces and Dev Containers, we need to handle the case where the secrets might not be present and give the best messaging. Perhaps we could dynamically select which messaging to use based on whether the expected secret env vars are set?